docs/datagen/server/tags.md

```diff
@@ -0,0 +1,83 @@
+Tag Generation
+==============
+
+Tags can be generated for a mod by subclassing `TagsProvider` and implementing `#addTags`. After implementation, the provider must be [added][datagen] to the `DataGenerator`.
+
+`TagsProvider`
+--------------
+
+The tags provider has two methods used for generating tags: creating a tag with objects and other tags via `#tag`, or using tags from other object types to generate the tag data via `#getOrCreateRawBuilder`.
+
+!!! note
+    Typically, a provider will not call `#getOrCreateRawBuilder` directly unless a registry contains a representation of objects from a different registry (blocks have item representations to obtain the blocks in the inventory).
+
+When `#tag` is called, a `TagAppender` is created which acts as a chainable consumer of elements to add to the tag:
+
+Method           | Description
+:---:            | :---
+`add`            | Adds an object to a tag. 
+`addOptional`    | Adds an object to a tag through its name. If the object is not present, then the object will be skipped when loading.
+`addTag`         | Adds a tag to a tag. All elements within the inner tag are now a part of the outer tag.
+`addOptionalTag` | Adds a tag to a tag through its name. If the tag is not present, then the tag will be skipped when loading.
+`replace` | When `true`, all other entries added to this tag from other datapacks will be discarded.
+`remove` | Removes an object or tag from a tag.
+
+```java
+// In some TagProvider#addTags
+this.tag(EXAMPLE_TAG)
+  .add(EXAMPLE_OBJECT) // Adds an object to the tag
+  .addOptional(new ResourceLocation("othermod", "other_object")) // Adds an object from another mod to the tag
+
+this.tag(EXAMPLE_TAG_2)
+  .addTag(EXAMPLE_TAG) // Adds a tag to the tag
+  .remove(EXAMPLE_OBJECT) // Removes an object from this tag
+```
+
+!!! important
+    If the mod's tags softly depends on another mod's tags, the other mods' tags should be referenced using the optional methods.
```

Comment:
Perhaps it may be good to define what "softly depends" is, indirectly.
```suggestion
!!! important
    If the mod's tags softly depends on another mod's tags (the other mod may or may not be present at runtime), the other mods' tags should be referenced using the optional methods.
```